### PR TITLE
feat: tests on PR & push to main

### PR DESCRIPTION
**Pull request by ArzelaAscoIi**
in case you are trying to push a component that fails linting: 
image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -44,7 +44,7 @@ detached = true
 [tool.hatch.envs.dc.scripts]
 build = "mkdir -p dist && zip -r dist/custom_component.zip ./* -x 'dist/*' '**/__pycache__/*'"
 push = "curl --request POST --url https://api.cloud.deepset.ai/api/v2/custom_components --header 'accept: application/json' --header \"Authorization: Bearer $API_KEY\" --form 'file=@dist/custom_component.zip;type=application/zip'"
-build-and-push = "mkdir -p dist && zip -r dist/custom_component.zip ./* -x 'dist/*' '**/__pycache__/*' && curl --request POST --url https://api.cloud.deepset.ai/api/v2/custom_components --header 'accept: application/json' --header \"Authorization: Bearer $API_KEY\" --form 'file=@dist/custom_component.zip;type=application/zip'"
+build-and-push = "hatch run code-quality:all && mkdir -p dist && zip -r dist/custom_component.zip ./* -x 'dist/*' '**/__pycache__/*' && curl --request POST --url https://api.cloud.deepset.ai/api/v2/custom_components --header 'accept: application/json' --header \"Authorization: Bearer $API_KEY\" --form 'file=@dist/custom_component.zip;type=application/zip'"
```
Comment (ArzelaAscoIi) [inline pyproject.toml:47]: thats a bit drastic, but it should help us fixing bugs in dC

File: pyproject.toml [hunk 2]
```diff
@@ -63,6 +63,13 @@ dependencies = [
 packages = ["src/dc_custom_component"]
 
 [tool.hatch.envs.code-quality.scripts]
+format = "black src tests --check"
+format-fix = "black src tests"
+lint = "ruff ."
```
Comment (faymarie) [inline pyproject.toml:68]: I believe it was changed to `ruff check .`
```suggestion
lint = "ruff ."
```

Comment (ArzelaAscoIi) [reply pyproject.toml:68]: thats the lint-ci command below 🙈

Comment (faymarie) [reply pyproject.toml:68]: Don't we usually do `ruff check --output-format=github` for the lint-ci command?

Comment (ArzelaAscoIi) [reply pyproject.toml]: oh damn. you are right. it should also be ruff check . (--fix). What i used is deprecated

**Review by faymarie — APPROVED** [review]

**Pull request merged**